src/w2grid.js

```diff
@@ -2572,7 +2572,7 @@
                                 this.last.xhr_hasMore = true;
                             } else {
                                 this.last.xhr_hasMore = false;
-                                this.total = this.last.xhr_offset + data.records.length;
+                                this.total = this.offset + this.last.xhr_offset + data.records.length;
```

Comment:
Thank you for submitting the pull request. However, I am not sure that it is correct. I looked up in the code and I see that this.last.xhr_offset already has this.offset in it. Adding it second time would be a mistake. Also, this condition is only applicable in infinite scroll and when the number of records is not set from the server ahead of time.

Please provide a use case when that shows that this change is good. If you can create a sample service and an example on jsFiddle (or similar) it would be great.

Reply:
The number of records isn't always known ahead of time. One of the reasons is that data could still be in the process of loading/generating server-side as the user is scrolling.

An example of this behavior is in "steps to reproduce" in issue #1775 .